src/main/java/org/sonarsource/sonarlint/ls/SonarLintLanguageServer.java

```diff
@@ -172,6 +172,8 @@ static SonarLintLanguageServer bySocket(int port, Collection<URL> analyzers) thr
       String clientVersion = params.getClientInfo().getVersion();
       String ideVersion = appName + " " + clientVersion;
 
+      httpClient.setUserAgent(productName);
```

Comment:
As discussed, I would rename this method 'initialize' and create the internal Http client directly.